rmw/include/rmw/types.h

```diff
@@ -36,6 +36,18 @@ extern "C"
 // implementation. It may need to be increased in the future.
 #define RMW_GID_STORAGE_SIZE 24
 
+/// Define QoS policy events
+typedef enum rmw_event_type_t
```

Comment:
This is incomplete. Really RMW_EVENT_SUBSCRIPTION_MATCHED and RMW_EVENT_OFFERED_INCOMPATIBLE_QOS should be removed as well. @ross-desmond this affects your map lookup in rmw_connext though. Seem reasonable for this removal?